adapter-mongodb/src/main/java/io/nosqlbench/adapter/mongodb/dispensers/MongoCommandOpDispenser.java

```diff
@@ -30,40 +29,33 @@
 import java.util.function.LongFunction;
 
 public class MongoCommandOpDispenser extends BaseOpDispenser<Op, MongoSpace> {
-    private final LongFunction<MongoDirectCommandOp> opFunc;
+
     private final LongFunction<MongoDirectCommandOp> mongoOpF;
 
-    public MongoCommandOpDispenser(DriverAdapter adapter, LongFunction<MongoSpace> ctxFunc, ParsedOp op) {
-        super(adapter,op);
-        opFunc = createOpFunc(ctxFunc, op);
-        this.mongoOpF = createOpFunc(ctxFunc,op);
+    public MongoCommandOpDispenser(DriverAdapter<Op, MongoSpace> adapter, LongFunction<MongoSpace> ctxFunc, ParsedOp op) {
+        super(adapter, op);
+        this.mongoOpF = createOpFunc(ctxFunc, op);
     }
 
     private LongFunction<MongoDirectCommandOp> createOpFunc(LongFunction<MongoSpace> ctxFunc, ParsedOp op) {
 
-        LongFunction<String> rpstring = op.getAsOptionalFunction("readPreference")
-            .orElseGet(() -> op.getAsOptionalFunction("read-preference")
-                .orElse(l -> "primary"));
-        LongFunction<ReadPreference> readPreferenceF = l -> ReadPreference.valueOf(rpstring.apply(l));
-
         LongFunction<?> payload = op.getAsRequiredFunction("stmt", Object.class);
         Object exampleValue = payload.apply(0);
 
         LongFunction<Bson> bsonFunc;
-        if (exampleValue instanceof CharSequence cs) {
+        if (exampleValue instanceof CharSequence) {
```

Comment:
Code formatting on this is retrograde. The modern Java 17 syntax should be preferred everywhere.

Reply:
Correct, that isn't used anywhere for this particular logic.